lib/crawly/utils.ex

```diff
@@ -174,6 +174,65 @@ defmodule Crawly.Utils do
     )
   end
 
+  @doc """
+  General purpose requests extractor
+
+  Takes html body of the page, base_url and a list of url filters and produces
+  new requests to follow.
+
+  The code below extracts all links from a page, and tries to match each of them
+  with a list of filters, making a request from each matched link.
+
+  NOTE: This function is created in order to spiders creation, however it's less
+  performant comparing to the approach when links are extracted directly (e.g.
+  from the pagination, etc). So to summarize, it's slow, but in most cases that
+  will not be noticeable.
+  """
+  @spec extract_requests(page_body, base_url, list_of_filters) :: result
+        when page_body: String.t(),
+             base_url: String.t(),
+             list_of_filters: String.t(),
+             result: [Crawly.Request.t()]
+  def extract_requests(page_body, base_url, filters) do
```

Comment:
I think that this entire function should be placed inside the `Crawly.Parsers.ExtractRequests` module, with the main logic inside the `run/3` and perhaps a convenience function for people to use it in `parse_items/1` if they want to.

Reply:
The way i see the `Utils` module is for commonly used utility functions that are critically used within the Engine and also for Spiders and Pipelines, so that things conform to what the Engine/Manager/Worker expects.